src/graph/graph.h

```diff
@@ -87,15 +87,21 @@ void Graph_AcquireReadLock(Graph *g);
 /* Acquire a lock for exclusive access to this graph's data */
 void Graph_AcquireWriteLock(Graph *g);
 
+/* Release a read lock and acquire a write lock without modifying the '_writelocked' flag. */
+void Graph_UpgradeTemporaryLock(Graph *g);
+
+/* Release the held lock */
+void Graph_ReleaseLock(Graph *g);
+
+/* Release the held lock without modifying the '_writelocked' flag.  */
+void Graph_ReleaseTemporaryLock(Graph *g);
```

Comment:
Our locking logic is complex enough without these additions, can we use the graph RWLock without these additions?
I feel like they make this entire process more complex, if the answer is no, then I would prefer introducing a new mutex guarding the attribute rax and avoid mixing the graph lock with the attribute storage